core/src/main/java/io/debezium/connector/cassandra/RangeData.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.connector.cassandra;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+
+import org.apache.kafka.connect.data.Schema;
+import org.apache.kafka.connect.data.SchemaBuilder;
+import org.apache.kafka.connect.data.Struct;
+import org.apache.kafka.connect.errors.DataException;
+
+import io.debezium.DebeziumException;
+
+public class RangeData implements KafkaRecord {
+
+    public final String name;
+    public final String method;
+    public final Map<String, String> values = new HashMap<>();
+
+    public RangeData(String name, String method, Map<String, String> values) {
+        this.name = name;
+        this.method = method;
+        if (values != null) {
+            this.values.putAll(values);
+        }
+    }
+
+    public static RangeData start(String method, Map<String, String> values) {
+        return new RangeData("_range_start", method, values);
+    }
+
+    public static RangeData end(String method, Map<String, String> values) {
+        return new RangeData("_range_end", method, values);
+    }
+
+    @Override
+    public Struct record(Schema schema) {
+        try {
+            return new Struct(schema)
+                    .put("method", method)
+                    .put("values", values);
+        }
+        catch (DataException e) {
+            throw new DebeziumException(e);
+        }
+    }
+
+    static Schema rangeSchema(String name) {
+        return SchemaBuilder.struct().name(name)
+                .field("method", Schema.STRING_SCHEMA)
+                .field("values", SchemaBuilder.map(Schema.STRING_SCHEMA, Schema.STRING_SCHEMA))
```

Comment:
IMHO the schema here is not correct, you still need to call build after the `map`. Again symobilc name and version number would preferred and created in the schema factory centralized class.

Reply:
@jpechane could you point me to example of these versions and names? I am not completely sure I follow here.

Reply:
ah ok I see what you mean